clevertap-core/src/main/java/com/clevertap/android/sdk/network/http/UrlConnectionHttpClient.kt

```diff
@@ -0,0 +1,101 @@
+            Logger.d("Pinning SSL session to DigiCertGlobalRoot CA certificate")
+            sslContext?.socketFactory
+        } catch (e: Exception) {
+            Logger.d("Issue in pinning SSL,", e)
+            null
+        }
+    }
+    private val sslContext: SSLContext? by lazy { createSslContext() }
+
+    override fun execute(request: Request): Response {
+        var connection: HttpsURLConnection? = null
+
+        try {
+            connection = openHttpsURLConnection(request)
+
+            if (request.body != null) {
+                connection.doOutput = true
+                connection.outputStream.use {
+                    it.write(request.body.toByteArray(Charsets.UTF_8))
+                }
+            }
+            logger.debug(logTag, "Sending request to: ${request.url}")
+
+            // execute request
+            val responseCode = connection.responseCode
+            val headers = connection.headerFields
+            val disconnectConnection = { connection.disconnect() }
+
+            return if (responseCode == HttpURLConnection.HTTP_OK) {
+                Response(request, responseCode, headers, connection.inputStream, disconnectConnection)
+            } else {
+                Response(request, responseCode, headers, connection.errorStream, disconnectConnection)
+            }
+        } catch (e: Exception) {
+            connection?.disconnect()
+            throw e
+        }
+    }
+
+    private fun openHttpsURLConnection(request: Request): HttpsURLConnection {
+        val url = URL(request.url.toString())
+        val connection = url.openConnection() as HttpsURLConnection
+        connection.connectTimeout = connectTimeout
+        connection.readTimeout = readTimeout
+        for (header in request.headers) {
+            connection.setRequestProperty(header.key, header.value)
+        }
+        connection.instanceFollowRedirects = false
+        if (isSslPinningEnabled && sslContext != null) {
+            connection.sslSocketFactory = sslSocketFactory
+        }
+        return connection
+    }
+
+    private fun createSslContext(): SSLContext? {
+        try {
+            val certificateFactory = CertificateFactory.getInstance("X.509")
+            val trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm())
+            val keyStore = KeyStore.getInstance(KeyStore.getDefaultType())
+            keyStore.load(null, null) //Use null InputStream & password to create empty key store
+            val inputStream: InputStream =
+                BufferedInputStream(javaClass.classLoader?.getResourceAsStream("com/clevertap/android/sdk/certificates/AmazonRootCA1.cer"))
+            val x509Certificate3 = certificateFactory.generateCertificate(inputStream) as X509Certificate
+            keyStore.setCertificateEntry("AmazonRootCA1", x509Certificate3)
+            trustManagerFactory.init(keyStore)
+            val sslContext = SSLContext.getInstance("TLS")
```

Comment:
There is a Snyk warning on this line:` Implement secure HTTPS communication. Consider using TLSv1.2 instead`. 

The suggested fix will not do anything useful. 
SSLContext.getInstance("TLS") returns an instance which supports different protocols depending on the Android API version. For api 19-25 this is ["SSLv3", "TLSv1", "TLSv1.1", "TLSv1.2"] and 26+ is ["TLSv1", "TLSv1.1", "TLSv1.2", "TLSv1.3"]. 
If we change this to SSLContext.getInstance("TLSv1.2") as the fix suggests we will only remove version 1.3 and up from the supported protocols and the older ones will still be available, which defeats the purpose of the warning.

We could 1) create an ssl factory that explicitly supports only TLS 1.2 and 1.3 since the backend support them or 2) disable the warning and keep this as it is. Currently if the backend and device both support TLS 1.2/1.3 they are used

Reply:
Lets disable the warning for now

Reply:
@vasct @CTLalit I will check how we can disable/bypass this for the time being

Reply:
Todor is going to assist us with bypassing this. I will inform @vasct once we have this done